lib/widgets/home.dart

```diff
@@ -2,96 +2,528 @@ import 'package:flutter/material.dart';
-          foregroundColor: WidgetStatePropertyAll(colorScheme.onSecondaryContainer))),
-        child: Center(
-          child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
-            DefaultTextStyle.merge(
-              textAlign: TextAlign.center,
-              style: const TextStyle(fontSize: 18),
-              child: Column(children: [
-                Text.rich(TextSpan(
-                  text: 'Connected to: ',
-                  children: [bold(store.realmUrl.toString())])),
-              ])),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                MessageListPage.buildRoute(context: context,
-                  narrow: const CombinedFeedNarrow())),
-              child: Text(zulipLocalizations.combinedFeedPageTitle)),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                MessageListPage.buildRoute(context: context,
-                  narrow: const MentionsNarrow())),
-              child: Text(zulipLocalizations.mentionsPageTitle)),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                MessageListPage.buildRoute(context: context,
-                  narrow: const StarredMessagesNarrow())),
-              child: Text(zulipLocalizations.starredMessagesPageTitle)),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                InboxPage.buildRoute(context: context)),
-              child: const Text("Inbox")), // TODO(i18n)
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                SubscriptionListPage.buildRoute(context: context)),
-              child: const Text("Subscribed channels")),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                RecentDmConversationsPage.buildRoute(context: context)),
-              child: Text(zulipLocalizations.recentDmConversationsPageTitle)),
-            if (testStreamId != null) ...[
-              const SizedBox(height: 16),
-              ElevatedButton(
-                onPressed: () => Navigator.push(context,
-                  MessageListPage.buildRoute(context: context,
-                    narrow: ChannelNarrow(testStreamId!))),
-                child: const Text("#test here")), // scaffolding hack, see above
-            ],
+      appBar: AppBar(),
+      body: const LoadingPlaceholder(),
+      floatingActionButton: TextButton(
+        onPressed: () => Navigator.push(context,
+          MaterialWidgetRoute(page: const ChooseAccountPage())),
+        child: Text(zulipLocalizations.chooseAccountPageTitle)));
+  }
```

Comment:
Yeah, "Choose account" isn't good text here. You made the same point about the button in the menu and I agree. Better text would be "Switch account", or perhaps like "Try another account".

The reason something like this is needed is in case the data for this account takes a long time to load — perhaps isn't ever going to load, because the server is down — so the user can still use some other account if that's what they wanted.

Reply:
In addition to using "Try another account", does it look better as a `FilledButton` or an `OutlinedButton`? (so that user knows that it is a button)

| `TextButton` | `OutlinedButton` | `FilledButton` |
| - | - | - |
| ![Screenshot from 2024-12-06 15-09-11](https://github.com/user-attachments/assets/ff882c7d-1c5f-4de8-a924-17154e5164ae) | ![Screenshot from 2024-12-06 15-09-03](https://github.com/user-attachments/assets/88073307-90d5-41e9-8a92-c5dc1b60e835) | ![Screenshot from 2024-12-06 15-08-51](https://github.com/user-attachments/assets/202d855e-78e4-4dac-ae86-9d3d5cae464f) |

Reply:
Yeah, I think the button's current appearance doesn't really read as a button.

Maybe a nicer alternative (might take longer to implement) would be to wait 5 seconds or so before showing anything, and have it be a message (centered in the page?) like "This is taking longer than usual to load. Try another account?" That way in the happy-path case we don't distract the user by making them figure out UI that they won't need.

Reply:
Started a discussion [here](https://chat.zulip.org/#narrow/channel/48-mobile/topic/.23F1076.20accessing.20choose-account.20page.20from.20the.20loading.20screen).